src/main/java/org/truffleruby/Main.java

```diff
@@ -88,42 +85,57 @@ public static void main(String[] args) {
             System.exit(mee.getStatus());
         }
 
+        final int exitCode;
+
+        final String filename = config.getDisplayedFileName();
+
+        config.setOption(
+                OptionsCatalog.ORIGINAL_INPUT_FILE,
+                config.shouldUsePathScript() ? config.getScriptFileName() : filename);
+        config.setOption(
+                OptionsCatalog.GRAAL_PRESENT,
+                String.valueOf(Truffle.getRuntime().getName().toLowerCase(
+                        Locale.ENGLISH).contains("graal")));
```

Comment:
Ruby should be able to read it from Main and future launcher, I've removed the options.